src/service/usres.py

```diff
@@ -0,0 +1,28 @@
+class UserService:
+
+    def __init__(self, dbcursor):
+        self.dbcursor = dbcursor
+
+    def create(self, name, email, password, created_at, connection):
+        self.dbcursor.execute(f"INSERT INTO `users`(`usr_name`, `usr_email`, `usr_encrypted_password`, `usr_created_at`) VALUES ('{name}','{email}','{password}','{created_at}')")
+        connection.commit()
+
+    def list_users(self):
+        self.dbcursor.execute("SELECT usr_id,usr_name, usr_email, usr_created_at, usr_encrypted_password FROM `users`;")
+        return self.dbcursor.fetchall()
+
+    def get_user_by_id(self,usr_id):
+        self.dbcursor.execute(f"SELECT * FROM users WHERE usr_id = {usr_id}")
+        return self.dbcursor.fetchall()[0]
+
+    def update_user(self, name, email, user_id, connection, password_form):
+        if password_form == 0:
```

Comment:
deve haver uma forma melhor de verificar se a senha está vazia. 
if password_form == 0: 
pode ser que se a variavel password_form form uma string vazia, password_form = '' seu codigo buga.

Reply:
eu fiz esse tratamento no serviço mano

```
    pass_form = request.form['password']
    if len(pass_form) == 0: 
        new_pass_form = 0
    else:
        new_pass_form = hashlib.sha256(pass_form.encode()).hexdigest()
```

Ou seja se o password_form for uma string vazia eu transformo ele em zero e ai se o valor for igual a 0 eu não altero o valor da senha, altero somente os demais dados do usuário.